fairlearn/metrics/__init__.py

```diff
@@ -18,81 +18,71 @@
-from ._skm_wrappers import group_max_error  # noqa: F401
-from ._skm_wrappers import group_mean_absolute_error  # noqa: F401
-from ._skm_wrappers import group_mean_squared_error  # noqa: F401
-from ._skm_wrappers import group_mean_squared_log_error  # noqa: F401
-from ._skm_wrappers import group_median_absolute_error  # noqa: F401
-from ._skm_wrappers import group_root_mean_squared_error  # noqa: F401
-from ._skm_wrappers import group_r2_score  # noqa: F401
-
-from ._metrics_engine import make_group_metric, group_summary  # noqa: F401
-from ._metrics_engine import group_min_from_summary, group_max_from_summary  # noqa: F401
-from ._metrics_engine import difference_from_summary, ratio_from_summary  # noqa: F401
-
-# -------------------------------------------
+from ._extra_metrics import (  # noqa: F401
+    true_positive_rate,
+    true_negative_rate,
+    false_positive_rate,
+    false_negative_rate,
+    balanced_root_mean_squared_error,
+    mean_overprediction,
+    mean_prediction,
+    mean_underprediction,
+    selection_rate)
+
+from ._metrics_engine import (  # noqa: F401
+    make_metric_group_summary, group_summary,
+    make_derived_metric,
+    group_min_from_summary, group_max_from_summary,
+    difference_from_summary, ratio_from_summary,
+    _metric_group_summary_dict, _derived_metric_dict)
+
+from ._disparities import (  # noqa: F401
+    demographic_parity_difference,
+    demographic_parity_ratio,
+    equalized_odds_difference,
+    equalized_odds_ratio,
+)
+
 
 _extra_metrics = [
+    "true_positive_rate",
+    "true_negative_rate",
+    "false_positive_rate",
+    "false_negative_rate",
     "balanced_root_mean_squared_error",
-    "fallout_rate",
     "mean_prediction",
     "mean_overprediction",
     "mean_underprediction",
-    "miss_rate",
     "selection_rate",
-    "specificity_score"
 ]
 
-_group_metrics = [
-    "group_accuracy_score",
-    "group_balanced_root_mean_squared_error",
-    "group_confusion_matrix",
-    "group_fallout_rate",
-    "group_max_error",
-    "group_mean_absolute_error",
-    "group_mean_prediction",
-    "group_mean_overprediction",
-    "group_mean_squared_error",
-    "group_mean_squared_log_error",
-    "group_mean_underprediction",
-    "group_median_absolute_error",
-    "group_miss_rate",
-    "group_precision_score",
-    "group_r2_score",
-    "group_recall_score",
-    "group_roc_auc_score",
-    "group_root_mean_squared_error",
-    "group_selection_rate",
-    "group_specificity_score",
-    "group_zero_one_loss"
-]
 
-_engine = [
-    "GroupMetricResult",
-    "make_group_metric",
+_metrics_engine = [
+    "make_metric_group_summary",
     "group_summary",
+    "make_derived_metric",
     "group_min_from_summary",
     "group_max_from_summary",
     "difference_from_summary",
     "ratio_from_summary"
 ]
 
 
-__all__ = _engine + _extra_metrics + _group_metrics
+globals().update(_metric_group_summary_dict)
```

Comment:
why is this required?

Reply:
This adds all the metrics of the form `<metric>_group_summary` to the package module and similarly the next line adds `<metric>_{difference,ratio,group_min,group_max}`. I'll add a comment to explain. Let me know if you are aware of a more idiomatic way to do this. (But that could be done in a future PR too.)

Reply:
So this is to avoid listing all of them individually?
I've never seen this before, and I wonder whether there are any negative consequences of doing this.
Perhaps @adrinjalali has thoughts?